contracts/orbital-core/src/utils.rs

```diff
@@ -0,0 +1,83 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{ensure, Coin, MessageInfo, StdError, StdResult, Uint128, Uint64};
+use cw_utils::must_pay;
+use neutron_sdk::bindings::msg::IbcFee;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::error::ContractError;
+
+pub fn assert_fee_payment(info: &MessageInfo, expected_fee: &Coin) -> Result<(), ContractError> {
+    match must_pay(info, &expected_fee.denom) {
+        Ok(amt) => ensure!(
+            amt >= expected_fee.amount,
+            ContractError::DomainRegistrationError("insufficient fee".to_string())
+        ),
+        Err(_) => {
+            return Err(ContractError::DomainRegistrationError(format!(
+                "no funds sent; expected {}.",
+                expected_fee
+            )))
+        }
```

Comment:
I'd actually return the PaymentError here, because the cause could be multiple reasons:
- No funds sent
- Wrong funds sent
- Multiple funds sent

Reply:
totally. updated `assert_fee_payment`! kinda wish `PaymentError` had a variant for `InsufficientFunds` or something, but just returning a `DomainRegistrationError` for that case for now.